### PR TITLE
[config] add `extra_tags` setting

### PR DESCRIPTION
**Pull request by vickenty**
### What does this PR do?

`extra_tags` specifies additional tags that will be appended to the host tags. Allows tags to be set via both the configuration file and environment variable at the same time. Duplicates will be removed.

### Describe your test plan

Provide some tags in `datadog.yaml` and set `DD_EXTRA_TAGS` to a different set of tags. Verify that tags from both sources are being sent.

### CODE REVIEW

File: pkg/metadata/host/host_tags.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ func GetHostTags(cached bool) *Tags {
 	}
 
 	t := &Tags{
-		System:              hostTags,
+		System:              util.SortUniqInPlace(hostTags),
```
Comment (ogaca-dd) [inline pkg/metadata/host/host_tags.go:148]: What do you think about moving the call `util.SortUniqInPlace` inside `GetConfiguredTags`?

**Review by ogaca-dd — COMMENTED** [review]
LGTM!

Note: I think it worth adding a section inside `pkg/config/config_template.yaml`.

File: pkg/metadata/host/host_tags.go (hunk 1 above)
Comment (vickenty) [reply pkg/metadata/host/host_tags.go:148]: This way de-duplication would only apply to configured tags before splitting, while duplicates can still occur due to tag splitting via `tag_value_split_separator`, and by having redundant or overlapping tags from k8s/gke/ec2.

**Comment by vickenty** [comment]
> Note: I think it worth adding a section inside `pkg/config/config_template.yaml`.

Added. Any other documentation that needs updating?

**Review by sarina-dd — COMMENTED** [review]
👍  LGTM release notes

**Review by ogaca-dd — APPROVED** [review]

**Review by gbbr — APPROVED** [review]

**Comment by clamoriniere** [comment]
Hi @vickenty 
I think this feature is useful, but maybe we should to it a bit differently.
My concerne is: the new `extra_tags` can also be part of the configuration. so at one point someone will ask for `extra_extra_tags` because they are using `tags` and `extra_tags` with two different deployment template tools, and so you will need another extra variable for providing tags coming from env var.

Maybe `extra_tags` should only be configurable with an env-var and not with the configuration file. And be more clear `extra_tags` could be rename `extra_tags_from_env`.
WDYT?

**Review by clamoriniere — COMMENTED** [review]
Hi @vickenty 
I think this feature is useful, but maybe we should to it a bit differently.
My concerne is: the new `extra_tags` can also be part of the configuration. so at one point someone will ask for `extra_extra_tags` because they are using `tags` and `extra_tags` with two different deployment template tools, and so you will need another extra variable for providing tags coming from env var.

Maybe `extra_tags` should only be configurable with an env-var and not with the configuration file. And be more clear `extra_tags` could be rename `extra_tags_from_env`.
WDYT?

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -67,6 +67,16 @@ api_key:
 #   - environment:dev
 #   - <TAG_KEY>:<TAG_VALUE>
 
+## @param extra_tags - list of key:value elements - optional
+## List of aditional host tqgs. Attached in-app to every metric, event, log,
```
Comment (lucaspimentel) [inline pkg/config/config_template.yaml:71]:
```suggestion
## List of aditional host tags. Attached in-app to every metric, event, log,
```

**Comment by vickenty** [comment]
> My concerne is: the new `extra_tags` can also be part of the configuration. so at one point someone will ask for `extra_extra_tags` because they are using `tags` and `extra_tags` with two different deployment template tools, and so you will need another extra variable for providing tags coming from env var.

Interesting point, I can definitely see some confusion here. I was thinking that customers that use `DD_TAGS` today would be able to add tags via configuration file without having to update their deployments, but I don't have a strong opinion here. Out of curiosity, is it possible to use two different deployment tools to set tags in the currently released agent?

**Comment by vickenty** [comment]
@clamoriniere I discussed this with the team, and we still want this to be available as a configuration file option, even if undocumented. I have removed `extra_tags` from the config template to avoid possible confusion, and instead it is mentioned as an environment variable in the `tags` comment. Does this address your concern?

**Review by clamoriniere — APPROVED** [review]
Thanks for the update. 
It looks ok to me.

**Review by sarina-dd — APPROVED** [review]

**Pull request merged**